sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeColumnCommand.scala

```diff
@@ -0,0 +1,168 @@
+
+
+/**
+ * Analyzes the given columns of the given table in the current database to generate statistics,
+ * which will be used in query optimizations.
+ */
+case class AnalyzeColumnCommand(
+    tableIdent: TableIdentifier,
+    columnNames: Seq[String]) extends RunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val sessionState = sparkSession.sessionState
+    val relation = EliminateSubqueryAliases(sessionState.catalog.lookupRelation(tableIdent))
+
+    // check correctness of column names
+    val attributesToAnalyze = mutable.MutableList[Attribute]()
+    val caseSensitive = sessionState.conf.caseSensitiveAnalysis
+    columnNames.foreach { col =>
+      val exprOption = relation.output.find { attr =>
+        if (caseSensitive) attr.name == col else attr.name.equalsIgnoreCase(col)
+      }
+      val expr = exprOption.getOrElse(throw new AnalysisException(s"Invalid column name: $col."))
+      // do deduplication
+      if (!attributesToAnalyze.contains(expr)) {
+        attributesToAnalyze += expr
+      }
+    }
+
+    relation match {
+      case catalogRel: CatalogRelation =>
+        updateStats(catalogRel.catalogTable,
+          AnalyzeTableCommand.calculateTotalSize(sparkSession, catalogRel.catalogTable))
+
+      case logicalRel: LogicalRelation if logicalRel.catalogTable.isDefined =>
+        updateStats(logicalRel.catalogTable.get, logicalRel.relation.sizeInBytes)
+
+      case otherRelation =>
+        throw new AnalysisException("ANALYZE TABLE is not supported for " +
+          s"${otherRelation.nodeName}.")
+    }
+
+    def updateStats(catalogTable: CatalogTable, newTotalSize: Long): Unit = {
+      // Collect statistics per column.
+      // The first element in the result will be the overall row count, the following elements
+      // will be structs containing all column stats.
+      // The layout of each struct follows the layout of the ColumnStats.
+      val ndvMaxErr = sessionState.conf.ndvMaxError
+      val expressions = Count(Literal(1)).toAggregateExpression() +:
+        attributesToAnalyze.map(ColumnStatsStruct(_, ndvMaxErr))
+      val namedExpressions = expressions.map(e => Alias(e, e.toString)())
+      val statsRow = Dataset.ofRows(sparkSession, Aggregate(Nil, namedExpressions, relation))
+        .queryExecution.toRdd.collect().head
+
+      // unwrap the result
+      val rowCount = statsRow.getLong(0)
+      val columnStats = attributesToAnalyze.zipWithIndex.map { case (expr, i) =>
+        (expr.name, ColumnStatsStruct.unwrapStruct(statsRow, i + 1, expr))
+      }.toMap
+
+      val statistics = Statistics(
+        sizeInBytes = newTotalSize,
+        rowCount = Some(rowCount),
+        colStats = columnStats ++ catalogTable.stats.map(_.colStats).getOrElse(Map()))
+      sessionState.catalog.alterTable(catalogTable.copy(stats = Some(statistics)))
+      // Refresh the cached data source table in the catalog.
+      sessionState.catalog.refreshTable(tableIdent)
```

Comment:
This is a bug. `tableIdent` might not have `database`. We need to qualify it. There are a couple of ways to do it.